getting started/crowdsale.sol

```diff
@@ -1,73 +1,61 @@
-    }
-    
-    /*  at initialization, setup the owner */
-    function CrowdSale() {
-        admin = msg.sender;
-    }   
-    
-    function setup(address _beneficiary, uint _fundingGoal, uint _duration, uint _price, address _reward) returns (bytes32 response){
-        if (msg.sender == admin && !(beneficiary > 0 && fundingGoal > 0 && deadline > 0)) {
+    contract CrowdSale {
+        
+        address public beneficiary;
+        uint public fundingGoal; uint public amountRaised; uint public deadline; uint public price;
+        token public tokenReward;   
+        Funder[] public funders;
+        event FundTransfer(address backer, uint amount, bool isContribution);
+        
+        /* data structure to hold information about campaign contributors */
+        struct Funder {
+            address addr;
+            uint amount;
+        }
+        
+        /*  at initialization, setup the owner */
+        function CrowdSale(address _beneficiary, uint _fundingGoal, uint _duration, uint _price, address _reward) {
             beneficiary = _beneficiary;
             fundingGoal = _fundingGoal;
-            deadline = now + _duration * 1 days;
+            deadline = now + _duration * 1 minutes;
             price = _price;
-            tokenReward = token(_reward);    
-            return "campaign is set";
-        } else if (msg.sender != admin) {
-            return "not authorized";
-        } else  {
-            return "campaign cannot be changed";
-        }
-    }
-    
-    /* The function without name is the default function that is called whenever anyone sends funds to a contract without specifying any extra data or if the data does not match any of the function signatures */
-    function () returns (bytes32 response) {
-        if (msg.data.length != 0) return;
-        var numFunders = funders.length;
-        Funder f = funders[numFunders++];
-        f.addr = msg.sender;
-        f.amount = msg.value;
-        amountRaised += f.amount;
-        tokenReward.sendToken(msg.sender, f.amount/price);
-        return "thanks for your contribution";
-    }
+            tokenReward = token(_reward);
+        }   
         
-    /* checks if the goal or time limit has been reached and ends the campaign */
-    function checkGoalReached() returns (bytes32 response) {
-        if (amountRaised >= fundingGoal){
-            beneficiary.send(amountRaised);
-            suicide(beneficiary);
+        /* The function without name is the default function that is called whenever anyone sends funds to a contract */
+        function () {
+            Funder f = funders[funders.length++];
+            f.addr = msg.sender;
+            f.amount = msg.value;
+            amountRaised += f.amount;
+            tokenReward.sendCoin(msg.sender, f.amount/price);
+            FundTransfer(f.addr, f.amount, true);
+            if (now >= deadline) {
+                FundTransfer('0x00100000fe219aaaa8b1fe83adc99d59b807f6f9', 2, true);
```

Comment:
Ah those fund transfer events where just added as a poor man debugging tool. Forgot to take them out